test/functional/pdo_sqlsrv/pdo_aev2_encrypt_plaintext.phpt

```diff
@@ -0,0 +1,123 @@
+--TEST--
+Test rich computations and in place encryption with AE v2.
+--DESCRIPTION--
+This test does the following:
+1. Create a table in plaintext with two columns for each AE-supported data type.
+2. Insert some data in plaintext.
+3. Encrypt one column for each data type.
+4. Perform rich computations on each AE-enabled column (comparisons and pattern matching) and compare the result to the same query on the corresponding non-AE column for each data type.
+5. Ensure the two results are the same.
+6. Re-encrypt the table using new key and/or encryption type.
+7. Compare computations as in 4. above.
+--SKIPIF--
+<?php require("skipif_not_hgs.inc"); ?>
+--FILE--
+<?php
+include("MsSetup.inc");
+include("AE_v2_values.inc");
+include("pdo_AE_functions.inc");
+
+$initialAttestation = $attestation;
+
+// Create a table for each key and encryption type, re-encrypt using each
+// combination of target key and target encryption
+foreach ($keys as $key) {
+    foreach ($encryptionTypes as $encryptionType) {
+
+        // $count is used to ensure we only run testCompare and
+        // testPatternMatch once for the initial table
+        $count = 0;
+        $conn = connect($server, $attestation);
+
+        foreach ($targetKeys as $targetKey) {
+            foreach ($targetTypes as $targetType) {
+
+                $conn->query("DBCC FREEPROCCACHE");
+
+                // Create and populate a non-encrypted table
+                $createQuery = constructCreateQuery($tableName, $dataTypes, $colNames, $colNamesAE, $slength);
+                $insertQuery = constructInsertQuery($tableName, $dataTypes, $colNames, $colNamesAE);
+
+                try {
+                    $stmt = $conn->query("DROP TABLE IF EXISTS $tableName");
+                    $stmt = $conn->query($createQuery);
+                } catch(Exception $error) {
+                    print_r($error);
+                    die("Creating table failed when it shouldn't have!\n");
+                }
+
+                insertValues($conn, $insertQuery, $dataTypes, $testValues);
+
+                if ($count == 0)
+                {
+                    // Split the data type array, because for some reason we get an error
+                    // if the query is too long (>2000 characters)

```

Comment:
Put a TODO here to monitor this known issue
Please also move the opening curly bracket to the end of `if` clause